src/components/socialMedia/SocialMediaContainer.js

```diff
@@ -13,11 +13,71 @@ export default function SocialMediaContainer(props){
     const [currentRace, setCurrentRace] = useState("unselected");
     const [profileCreated, setProfileCreated] = useState(false);
     const [refreshes, setRefreshes] = useState(0);
-    const [adFrequencies,] = useState(initialAds.adList); //setAdFrequencies & intervals to be added later
+    const [adFrequencies, setAdFrequencies] = useState(null); //setAdFrequencies & intervals to be added later
     const [selectedAd, setSelectedAd]= useState(null);
-    const [refreshing, setRefreshing] = useState(false);
+    const [adDataSet, setAdDataSet] = useState(-1);
 
+    //set adFrequencies when component mounts
+    useEffect(() => {
+        setAdFrequencies([...initialAds.adList]);
+    },[]);
 
+
+    //change dataSet to the correct one to examine based off of user's profile
+    //whenever profileCreated is changed
+    useEffect(() => {
+        if (profileCreated === false)
+            setAdDataSet(-1);
+        else if (currentGender === "male" && currentRace === "white")
+            setAdDataSet(0);
+        else if (currentGender === "male" && currentRace === "black")
+            setAdDataSet(1);
+        else if (currentGender === "female" && currentRace === "white")
+            setAdDataSet(2);
+        else if (currentGender === "female" && currentRace === "black")
+            setAdDataSet(3);
+        
+    }, [profileCreated, currentGender, currentRace]); //even though profileCreated, still including currentGender & currentRace in dependency array
+
+
+    function resetAds(){
+        setAdFrequencies( (prevFrequencies)=>{
+            //set all y values back to zero, reinitialize ad histogram
+            let newAdFrequencies = [...prevFrequencies];
+            let rV = newAdFrequencies.map(
+                (job) =>{
+                    return {...job, y : 0};
+                }
+            )
+            return rV;
+        });
+        setRefreshes(0);
+    }
+    function pickAd(value){
+        setAdFrequencies( (prevFrequencies) =>{
+            let newAdFrequencies = [...prevFrequencies];
+        //pick an ad the designated number of times
+        for (let i=0; i < value; i++){
+            let prob = Math.random();
+            let threshold=0;
+            //randomly select ad from dataset
+            for (let job of newAdFrequencies){
+                console.log("entered" + job.x);
+                threshold += parseFloat(job.adPercentages[adDataSet]);
```

Comment:
It seems like the order in which each job is looked at is not randomized, am I right? If that's the case, the selection process might be biased (the last job in the list would be least likely to be picked), so we'd need to randomize which job is looked at first.